File: JavaScript/1288-Remove Covered Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var removeCoveredIntervals = function(intervals) {
    intervals.sort(function(a,b){
        if (a[0]!==b[0]) return a[0]-b[0];
        else return b[1]-a[1];
    })
    
    let maxi = intervals[0][1];
    len = intervals.length;
    for (let i=1; i<intervals.length; i++){
        if (intervals[i][1] <= maxi) len-=1;
        else{
            maxi = Math.max(intervals[i][1], maxi);            
        }
    }
    return len;
};

File: C++/1288-Remove Covered Intervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(begin(intervals), end(intervals), [](auto &a, auto &b) {
            return a[0] < b[0] || (a[0] == b[0] && a[1] > b[1]);
        });
        int N = intervals.size();

        int res = 0, endPos = -1;

        for (int i=0; i<N; i++) {
            if (intervals[i][1] > endPos) {
                endPos = intervals[i][1];
                res++;
            }
        }

        return res;
    }
};